sys/include/tsrb.h

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_tsrb Thread safe ringbuffer
+ * @ingroup     sys
+ * @{
+ */
+
+/**
+ * @file
+ * @brief       Thread-safe ringbuffer implementation
+ *
+ * This ringbuffer implementation can be used without locking if
+ * there's only one producer and one consumer.
+ *
+ * @note Buffer size must be a power of two!
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef TSRB_H
+#define TSRB_H
+
+#include <assert.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief     thread-safe ringbuffer struct
+ */
+typedef struct tsrb {
+    char *buf;                  /**< Buffer to operate on. */
+    unsigned int size;          /**< Size of buf. */
+    volatile unsigned reads;    /**< total number of reads */
+    volatile unsigned writes;   /**< total number of writes */
+} tsrb_t;
+
+/**
+ * @brief Static initializer
+ */
+#define TSRB_INIT(BUF) { (BUF), sizeof (BUF), 0, 0 }
+
+/**
+ * @brief        Initialize a tsrb.
+ * @param[out]   rb        Datum to initialize.
+ * @param[in]    buffer    Buffer to use by tsrb.
+ * @param[in]    bufsize   `sizeof (buffer)`
+ */
+static inline void tsrb_init(tsrb_t *rb, char *buffer, unsigned bufsize)
+{
+    /* make sure bufsize is a power of two.
```

Comment:
could you also write this requirement somewhere into the doxygen block, so that it appears in the doc? Maybe using a `@pre` or `@note` command?